ibm/service/power/resource_ibm_pi_network_interface.go

```diff
@@ -0,0 +1,351 @@
+				Optional:    true,
+				Description: "The attached pvm-instance to this Network Interface.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_Href: {
+							Computed:    true,
+							Description: "Link to pvm-instance resource.",
+							Type:        schema.TypeString,
+						},
+						Attr_InstanceID: {
+							Computed:    true,
+							Description: "The attahed instance ID.",
+							Type:        schema.TypeString,
+						},
+					},
+				},
+				Type: schema.TypeList,
+			},
+			Attr_Status: {
+				Computed:    true,
+				Description: "The status of the network address group.",
+				Type:        schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceIBMPINetworkInterfaceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	networkID := d.Get(Arg_NetworkID).(string)
+	networkC := instance.NewIBMPINetworkClient(ctx, sess, cloudInstanceID)
+	var body = &models.NetworkInterfaceCreate{}
+	if v, ok := d.GetOk(Arg_IPAddress); ok {
+		body.IPAddress = v.(string)
+	}
+	if v, ok := d.GetOk(Arg_Name); ok {
+		body.Name = v.(string)
+	}
+	if v, ok := d.GetOk(Arg_UserTags); ok {
+		if len(flex.FlattenSet(v.(*schema.Set))) > 0 {
+			userTags := flex.FlattenSet(v.(*schema.Set))
+			body.UserTags = userTags
+		}
+	}
+	networkInterface, err := networkC.CreateNetworkInterface(networkID, body)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	networkInterfaceID := *networkInterface.ID
+	_, err = isWaitForIBMPINetworkInterfaceAvailable(ctx, networkC, networkID, networkInterfaceID, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if v, ok := d.GetOk(Arg_InstanceID); ok {
+		instanceID := v.(string)
+		body := &models.NetworkInterfaceUpdate{
+			InstanceID: &instanceID,
+		}
+		_, err = networkC.UpdateNetworkInterface(networkID, networkInterfaceID, body)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		_, err = isWaitForIBMPINetworkPortUpdateAvailable(ctx, networkC, networkID, networkInterfaceID, instanceID, d.Timeout(schema.TimeoutUpdate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+	d.SetId(fmt.Sprintf("%s/%s/%s", cloudInstanceID, networkID, networkInterfaceID))
+
+	return resourceIBMPINetworkInterfaceRead(ctx, d, meta)
+}
+
+func resourceIBMPINetworkInterfaceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	networkC := instance.NewIBMPINetworkClient(ctx, sess, parts[0])
+	networkInterface, err := networkC.GetNetworkInterface(parts[1], parts[2])
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.Set(Attr_CRN, networkInterface.Crn)
```

Comment:
Check for CRN not being set

Reply:
No need since it is not an optional field.